nibabel/brikhead.py

```diff
@@ -0,0 +1,488 @@
+           [ -0.    ,  -3.    ,  -0.    ,  82.312 ],
+           [  0.    ,   0.    ,   3.    , -52.3511],
+           [  0.    ,   0.    ,   0.    ,   1.    ]])
+    >>> head = load('example4d+orig.HEAD')
+    >>> np.array_equal(head.get_data(), brik.get_data())
+    True
+    """
+
+    header_class = AFNIHeader
+    valid_exts = ('.brik', '.head')
+    files_types = (('image', '.brik'), ('header', '.head'))
+    _compressed_suffixes = ('.gz', '.bz2')
+    makeable = False
+    rw = False
+    ImageArrayProxy = AFNIArrayProxy
+
+    @classmethod
+    @kw_only_meth(1)
+    def from_file_map(klass, file_map, mmap=True):
+        """
+        Parameters
+        ----------
+        file_map : dict
+            dict with keys ``image, header`` and values being fileholder
+            objects for the respective BRIK and HEAD files
+        mmap : {True, False, 'c', 'r'}, optional, keyword only
+            `mmap` controls the use of numpy memory mapping for reading image
+            array data.  If False, do not try numpy ``memmap`` for data array.
+            If one of {'c', 'r'}, try numpy memmap with ``mode=mmap``.  A
+            `mmap` value of True gives the same behavior as ``mmap='c'``.  If
+            image data file cannot be memory-mapped, ignore `mmap` value and
+            read array from file.
+        """
+        with file_map['header'].get_prepare_fileobj('rt') as hdr_fobj:
+            hdr = klass.header_class.from_fileobj(hdr_fobj)
+        imgf = file_map['image'].fileobj
+        if imgf is None:
+            imgf = file_map['image'].filename
+        data = klass.ImageArrayProxy(imgf, hdr.copy(), mmap=mmap)
+        return klass(data, hdr.get_affine(), header=hdr, extra=None,
+                     file_map=file_map)
+
+    @classmethod
+    @kw_only_meth(1)
+    def from_filename(klass, filename, mmap=True):
+        """
+        Parameters
+        ----------
+        file_map : dict
+            dict with keys ``image, header`` and values being fileholder
+            objects for the respective BRIK and HEAD files
+        mmap : {True, False, 'c', 'r'}, optional, keyword only
+            `mmap` controls the use of numpy memory mapping for reading image
+            array data.  If False, do not try numpy ``memmap`` for data array.
+            If one of {'c', 'r'}, try numpy memmap with ``mode=mmap``.  A
+            `mmap` value of True gives the same behavior as ``mmap='c'``.  If
+            image data file cannot be memory-mapped, ignore `mmap` value and
+            read array from file.
+        """
+        file_map = klass.filespec_to_file_map(filename)
+        # only BRIK can be compressed, but `filespec_to_file_map` doesn't
+        # handle that case; remove potential compression suffixes from HEAD
+        head_fname = file_map['header'].filename
+        if not os.path.exists(head_fname):
+            for ext in klass._compressed_suffixes:
+                head_fname = re.sub(ext, '', head_fname)
```

Comment:
I think this is going to remove `gz` or whatever inside filenames - for example - consider `head_fname = 'foogz_bar.nii.gz'`. 

I think you want something like:

```
for ext in klass._compressed_suffixes:
     head_fname = head_fname[:-len(ext)] if head_fname.endswith(ext) else head_fname
```